From 4487aab0503860021fba268a47a8dc20bd3020f6 Mon Sep 17 00:00:00 2001
From: Vadym Kochan <vadym.kochan@plvision.eu>
Date: Fri, 25 Oct 2019 16:02:25 +0300
Subject: [PATCH] package/iproute2: allow to build for kernel version < 3.4

Define __kernel_long_t type which was added in kernel 3.4, so iproute2
may compile for older kernel.

Signed-off-by: Vadym Kochan <vadym.kochan@plvision.eu>
---
 ...-support-compilation-for-Linux-versi.patch | 29 +++++++++++++++++++
 package/iproute2/Config.in                    |  4 +--
 2 files changed, 30 insertions(+), 3 deletions(-)
 create mode 100644 package/iproute2/0001-include-uapi-add-support-compilation-for-Linux-versi.patch

diff --git a/package/iproute2/0001-include-uapi-add-support-compilation-for-Linux-versi.patch b/package/iproute2/0001-include-uapi-add-support-compilation-for-Linux-versi.patch
new file mode 100644
index 0000000..f5569fd
--- /dev/null
+++ b/package/iproute2/0001-include-uapi-add-support-compilation-for-Linux-versi.patch
@@ -0,0 +1,29 @@
+From af985223f72788e4ecf9fba48f2cc60d37b1faea Mon Sep 17 00:00:00 2001
+From: Vadym Kochan <vadym.kochan@plvision.eu>
+Date: Fri, 25 Oct 2019 15:52:59 +0300
+Subject: [PATCH] include/uapi: add support compilation for Linux version < 3.4
+
+Signed-off-by: Vadym Kochan <vadym.kochan@plvision.eu>
+---
+ include/uapi/linux/sysinfo.h | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/include/uapi/linux/sysinfo.h b/include/uapi/linux/sysinfo.h
+index 435d5c2..d6ce0e4 100644
+--- a/include/uapi/linux/sysinfo.h
++++ b/include/uapi/linux/sysinfo.h
+@@ -4,6 +4,11 @@
+ 
+ #include <linux/types.h>
+ 
++#ifndef __kernel_long_t
++typedef long		__kernel_long_t;
++typedef unsigned long	__kernel_ulong_t;
++#endif
++
+ #define SI_LOAD_SHIFT	16
+ struct sysinfo {
+ 	__kernel_long_t uptime;		/* Seconds since boot */
+-- 
+2.17.1
+
diff --git a/package/iproute2/Config.in b/package/iproute2/Config.in
index ab9e515..6dd12c1 100644
--- a/package/iproute2/Config.in
+++ b/package/iproute2/Config.in
@@ -1,13 +1,11 @@
 config BR2_PACKAGE_IPROUTE2
 	bool "iproute2"
 	depends on BR2_USE_MMU # fork()
-	depends on BR2_TOOLCHAIN_HEADERS_AT_LEAST_3_4 # __kernel_{u,}long_t
 	help
 	  Kernel routing and traffic control utilities. Provides things
 	  like ip and tc.
 
 	  http://www.linuxfoundation.org/collaborate/workgroups/networking/iproute2
 
-comment "iproute2 needs a toolchain w/ headers >= 3.4"
+comment "iproute2 needs MMU"
 	depends on BR2_USE_MMU
-	depends on !BR2_TOOLCHAIN_HEADERS_AT_LEAST_3_4
-- 
2.17.1

